hrim.py

```diff
@@ -0,0 +1,98 @@
+import sys
+sys.path.insert(0, 'scripts')
+import printing
+import parsing
+import compiling
+import os
+from classes import *
+
+def main(args):
+	if args[1] == "show":
+		extend = False
+		# if topic/properties not defined on the parsed file should show their full content or not when printing
+		if len(args) == 4:
+			if args[3] == "1":
+				extend = True
+		# check for file input
+		if len(args) < 3:
+			print "Please input file to parse"
+			sys.exit()
+		module = parsing.main(args[2])
+		printing.main(module, extend)
+	elif args[1] == "generate":
+		extend = False
+		# check for file input
+		if len(args) < 3:
+			print "Please input file to parse"
+			sys.exit()
+		if args[2] == "all":
+			path = os.getcwd()
+			for file in [
+				path+"/models/composite/mobilebase/mobilebase.xml",
+				path+"/models/composite/conveyor/conveyor.xml",
+				path+"/models/composite/arm/arm.xml",
+				path+"/models/power/battery/battery.xml",
+				path+"/models/actuator/gripper/gripper.xml",
+				path+"/models/actuator/electricmotor/electricmotor.xml",
+				path+"/models/actuator/servo/servo.xml",
+				path+"/models/sensor/lidar/lidar.xml",
+				path+"/models/sensor/rangefinder/rangefinder.xml",
+				path+"/models/sensor/thermometer/thermometer.xml",
+				path+"/models/sensor/3dcamera/3dcamera_stereo.xml",
+				path+"/models/sensor/3dcamera/3dcamera_tof.xml",
+				path+"/models/sensor/3dcamera/3dcamera_depth.xml",
+				path+"/models/sensor/imu/imu.xml",
+				path+"/models/sensor/hygrometer/hygrometer.xml",
+				path+"/models/sensor/forcetorque/forcetorque.xml",
+				path+"/models/sensor/gps/gps.xml",
+				path+"/models/sensor/gasdetector/gasdetector.xml",
+				path+"/models/sensor/torque/torque.xml",
+				path+"/models/sensor/force/force.xml",
+				path+"/models/sensor/microphone/microphone.xml",
+				path+"/models/sensor/camera/camera.xml",
+				path+"/models/sensor/encoder/encoder.xml"
+
+			]:
+				os.chdir(path)
+				module = parsing.main(file)
+				compiling.main(module)
+		elif args[2] == "allClean":
+			path = os.getcwd()
+			for file in [
+				path+"/models/composite/mobilebase/mobilebase_clean.xml",
```

Comment:
Probably this way of adding folder is very handmade. Can you do something similar but adding all the folder below a directory.

Reply:
Implemented in https://github.com/erlerobot/HRIM/commit/928390266550c8e3eccd40fe218553ce2b02e2ad.

Currently the main [regex](https://github.com/erlerobot/HRIM/commit/928390266550c8e3eccd40fe218553ce2b02e2ad#diff-9ce68efd9817cf2f47a2e5a916019d8dR26) makes sure the "captured" **.xml** files are inside a categorization folder (i.e. sensor, actuator...) and aren't inside a "topic" folder (file separation for each module's own topics). It ignores OS differences as it checks for a single character instead of a slash/backslash, might be a bit hacky. Verified it also works on a Windows (7) system.

The check for [development files] is done checking for the filename ending in *_clean.xml*. This is because the [findModels](https://github.com/erlerobot/HRIM/commit/928390266550c8e3eccd40fe218553ce2b02e2ad#diff-9ce68efd9817cf2f47a2e5a916019d8dR12) method should filter everything but the models, both final and development ones.